### PR TITLE
Update the check point about libvirtd crash

### PR DESCRIPTION
**Pull request by yalzhang**
Currently when libvirtd crash, it will restart. So if we want to
check if there was a libvirtd crash, check the libvirtd status if
not enough, we should also check the pid of libvirtd.

Signed-off-by: Yalan Zhang <yalzhang@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/daemon/crash_regression.py [hunk 1]
```diff
@@ -54,9 +54,15 @@ def run_restart_save_restore(params, libvirtd, vm):
     Save and restore a domain after restart daemon.
     """
     libvirtd.restart()
+    ori_libvirtd_pid1 = process.getoutput("pidof libvirtd")
     save_path = os.path.join(data_dir.get_tmp_dir(), 'tmp.save')
     virsh.save(vm.name, save_path)
     virsh.restore(save_path)
+    if libvirtd.was_running:
+        aft_libvirtd_pid1 = process.getoutput("pidof libvirtd")
```
Comment (chunfuwen) [inline libvirt/tests/src/daemon/crash_regression.py:62]: We suggest use process.run() to get command ouput. 
why Process.getouput exists is to just  to keep code compatability.

Comment (yalzhang) [reply libvirt/tests/src/daemon/crash_regression.py:62]: Update, Thank you!

Comment (chunfuwen) [reply libvirt/tests/src/daemon/crash_regression.py:62]: it sounds like you miss the change for process.getoutput

File: libvirt/tests/src/daemon/crash_regression.py [hunk 2]
```diff
@@ -54,9 +54,15 @@ def run_restart_save_restore(params, libvirtd, vm):
     Save and restore a domain after restart daemon.
     """
     libvirtd.restart()
+    ori_libvirtd_pid1 = process.run("pidof libvirtd").stdout_text
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/crash_regression.py:57]: We'd better consider the situation when the split daemon is enabled, so I recommend that you use libvirtd.service_exec here...

File: libvirt/tests/src/daemon/crash_regression.py [hunk 3]
```diff
@@ -54,9 +54,15 @@ def run_restart_save_restore(params, libvirtd, vm):
...
     save_path = os.path.join(data_dir.get_tmp_dir(), 'tmp.save')
     virsh.save(vm.name, save_path)
     virsh.restore(save_path)
+    if libvirtd.was_running:
+        aft_libvirtd_pid1 = process.run("pidof libvirtd").stdout_text
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/crash_regression.py:62]: Same as above..

File: libvirt/tests/src/daemon/crash_regression.py [hunk 4]
```diff
@@ -236,6 +242,7 @@ def run(test, params, env):
     """
     Run various regression tests and check whether libvirt daemon crashes.
     """
+    global re
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/crash_regression.py:245]: May I ask why this line is added?

File: libvirt/tests/src/daemon/crash_regression.py [hunk 5]
```diff
@@ -255,10 +262,13 @@ def run(test, params, env):
                 shell=True, ignore_status=True)
     try:
         libvirtd.start()
+        ori_pid_libvirtd = process.run("pidof libvirtd").stdout_text
 
         run_func = globals()[func_name]
         for i in xrange(repeat):
-            run_func(params, libvirtd, vm)
+            re = run_func(params, libvirtd, vm)
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/crash_regression.py:269]: How about changing 're' to more meaningful name? like 'result', 'res'. I was thinking it as 're' module when I first saw it😅

Comment (qiankehan) [reply libvirt/tests/src/daemon/crash_regression.py:269]: > How about changing 're' to more meaningful name? like 'result', 'res'. I was thinking it as 're' module when I first saw itsweat_smile

Agree. ret make more sense

**Comment by qiankehan** [comment]
NACK. How do you know the libvirtd pid update is caused by libvirtd crash? It could be caused by libvirtd resrtart.
I think implementing **coredumpctl** with time range (_--since_ or _--until_) is better

**Comment by qiankehan** [comment]
What's more, libvirtd could crash in any cases. It's hard to prove that it is more likely to crash in these cases rather than the others.
I suggest to add this checking in avocado-vt and then apply to all the tp-libvirt cases.

**Comment by yalzhang** [comment]
As we have talked to use abrt to catch the libvirtd crash, close this PR

**Pull request closed without merging**